### PR TITLE
detect framework at wandb.init() time

### PR DESCRIPTION
**Pull request by raubitsj**
This is safer than the other approach since we do weird stuff writing to config with different processes :)

The old approach:
https://github.com/wandb/client/pull/550

### CODE REVIEW

File: wandb/__init__.py [hunk 1]
```diff
@@ -1034,11 +1034,14 @@ def set_global_config(run):
...
     run.config.set_run_dir(run.dir)
+    # we have re-read the config, add telemetry data
+    telemetry_updated = run.config._telemetry_update()
 
     if sagemaker_config:
         run.config._update(sagemaker_config)
         allow_val_change = True
-    if config:
+    if config or telemetry_updated:
+        print("about to save", config)
```
Comment (vanpelt) [inline wandb/__init__.py]: Print statement

Comment (raubitsj) [reply wandb/__init__.py]: oops, removed

**Comment by raubitsj** [comment]
Regression run:
https://app.wandb.ai/jeffr/regression/groups/20191101-0.8.13-feature_telemetry-imports-safer-0888399-9pe0zd

File: wandb/wandb_config.py [hunk 2]
```diff
@@ -217,6 +238,7 @@ def _sanitize_val(self, val):
             return val
 
     def _update(self, params, allow_val_change=False, as_defaults=False):
```
Comment (vanpelt) [inline wandb/wandb_config.py]: Do we want params=None here?

Comment (raubitsj) [reply wandb/wandb_config.py]: None might get passed from wandb.init(), before we were checking `if config:` but now we want to update telemetry also

**Pull request merged**